components/HTTP/src/Websocket.cpp

```diff
@@ -0,0 +1,303 @@
+{
+    req_handler = req;
+    int type = getTypeMessage(ws_pkt.payload);
+    WebsocketType r = static_cast<WebsocketType>(type);
+    client_data_t client;
+    esp_err_t err;
+    uid_message_t client_uid;
+    const char* null_to_uid = "ffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffffff";
+    bool update = false;
+
+    switch (r) {
+    case WebsocketType::Handshake:
+        ESP_LOGI(TAG, "!!!!!hanshake");
+        err = getClientData(ws_pkt.payload, &client);
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "type must be numeric 0 ");
+        };
+
+        if (m_client.size() > 0) {
+            int size = m_client.size();
+            for (size_t i = 0; i < size; i++) {
+                if (memcmp(m_client[i].shaUID, client.shaUID, sizeof(client.shaUID)) == 0) {
+                    m_client[i].fd = client.fd;
+                    update = true;
+                }
+            }
+            if (update == false) {
+                m_client.push_back(client);
+            }
+
+            update = false;
+        } else {
+            m_client.push_back(client);
+        }
+
+        std::cout << "vector: " << m_client.size() << std::endl;
+        break;
+    case WebsocketType::Message:
+        ESP_LOGI(TAG, "!!!!!msg");
+
+        err = messageRecipient(ws_pkt.payload, &client_uid, null_to_uid);
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "Error getting message uids %d", err);
+            return;
+        };
+
+        err = sendWsData(client_uid, ws_pkt, null_to_uid);
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "Error sending message %d", err);
+            return;
+        };
+
+        break;
+    case WebsocketType::Status:
+        ESP_LOGI(TAG, "!!!!!status");
+
+        err = messageRecipient(ws_pkt.payload, &client_uid, null_to_uid);
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "Error getting message uids %d", err);
+            return;
+        };
+
+        err = sendWsData(client_uid, ws_pkt, null_to_uid);
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "Error sending message %d", err);
+            return;
+        };
+
+        break;
+    case WebsocketType::Action:
+        ESP_LOGI(TAG, "!!!!!action");
+
+        err = messageRecipient(ws_pkt.payload, &client_uid, null_to_uid);
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "Error getting message uids %d", err);
+            return;
+        };
+
+        err = sendWsData(client_uid, ws_pkt, null_to_uid);
+        if (err != ESP_OK) {
+            ESP_LOGE(TAG, "Error sending message %d", err);
+            return;
+        };
+
+        break;
+    default:
+        break;
+    }
+}
+
+int Websocket::getTypeMessage(uint8_t* payload)
+{
+    char* res = reinterpret_cast<char*>(payload);
+    cJSON* req_root = cJSON_Parse(res);
+    cJSON* type = cJSON_GetObjectItemCaseSensitive(req_root, "type");
+    if (cJSON_IsNumber(type)) {
+        return type->valueint;
```

Comment:
```suggestion
        cJSON_Delete(req_root);
        return type->valueint;
```

We need to free cJSON memory after using it